content/livesync/models/index.textile

```diff
@@ -0,0 +1,203 @@
+- @eventBufferOptions@ := used to configure the in-memory sliding-window buffer used for reordering and deduplication.
+- @optimisticEventOptions@ := is used to configure how "optimistic events":/livesync/models/models#optimistic are applied.
+- @logLevel@ := configures the log level used to control the verbosity of log output. One of @fatal@, @error@, @warn@, @info@, @debug@, or @trace@.
+
+The following is an example of setting @ClientOptions@ when instantiating the Models SDK:
+
+```[javascript]
+const modelsClient = new ModelsClient({
+ client,
+ logLevel,
+ syncOptions: {
+  historyPageSize,
+  messageRetentionPeriod,
+  retryStrategy,
+ },
+ eventBufferOptions: {
+  bufferMs,
+  eventOrderer,
+ },
+ optimisticEventOptions: {
+  timeout,
+ },
+});
+```
+
+h2(#quickstart). Quickstart
+
+A model is a single instance of a live, observable data model backed by your database. In this guide, we will create a simple model that tracks a list of comments on a post.
+
+The state of our model will have the following type:
+
+```[javascript]
+type Post = {
+  id: number;
+  text: string;
+  comments: string[];
+};
+```
+
+To create the model, use the @models.get()@ method on the client. If a model with the given name already exists, it will be returned.
+
+To instantiate a Model you must provide a unique name. This identifies the model on the client, and is also the name of the "channel":/channels used to subscribe to state updates from the backend.
+
+```[javascript]
+const model = modelsClient.models.get({
+  name: 'myPost',
+  channelName: 'post:123',
+  sync,
+  merge,
+});
+```
+
+The model also requires:
+
+* a "sync function":/livesync/models/models#sync to initialize the model's state from the backend.
+* a "merge function":/livesync/models/models#merge to calculate the next version of the model state when change events are received from the backend.
+
+h3(#sync). Create a sync function
+
+Create a simple sync function which loads the post and its comments. The Models SDK will infer the type of the model state from the type of the data payload returned by the sync function.
+
+```[javascript]
+async function sync(id: number, page: number) {
+  const result = await fetch(`/api/post/${id}?page=${page}`);
+  return result.json();
+}
+```
+
+The following is an example response from the sync function:
+
+```[json]
+{
+  "sequenceID": "1",
+  "data": {
+    "id": 123,
+    "text": "Hello World",
+    "comments": []
+  }
+}
+```
+
+The sync endpoint on the backend returns the post data as well as a "@sequenceID@":/livesync/models/models#sequence which defines the point in the stream of change events that corresponds to this version of the data. You can obtain the @sequenceID@ by reading the largest @sequenceID@ from the "outbox table":/livesync/connector/tables#outbox-table in the same transaction that queries the post data.
+
+The following function will add a new comment using a backend endpoint:
```

Comment:
Lines 141-161 here should appear below "create a merge function"

Reply:
Oops, yep that was my mistake. Added in 5df7fc28f18f1034c61584cd428bdf64c99d867e